components/brave_today/rust/lib.rs

```diff
@@ -0,0 +1,121 @@
+extern crate feed_rs;
+use feed_rs::parser;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+#[cxx::bridge(namespace = brave_news)]
+mod ffi {
+    pub struct FeedItem {
+        id: String,
+        title: String,
+        description: String,
+        image_url: String,
+        destination_url: String,
+        published_timestamp: i64,
+    }
+
+    pub struct FeedData {
+        id: String,
+        title: String,
+        items: Vec<FeedItem>,
+    }
+
+    extern "Rust" {
+        fn parse_feed_string(source: String, output: &mut FeedData) -> bool;
```

Comment:
It would be nice if you could return a `Result<ffi::FeedData>` here instead, but `cxx` turns the `Err` branch into a C++ exception and we build with `-fno-exceptions` so that doesn't work. This seems like a reasonable approach; although the compiler can't check this, the partial-update issue with this function signature is handled reasonably here.